ddtrace/sampler.py

```diff
@@ -137,22 +138,30 @@ def set_sample_rate(
         # type: (...) -> None
         self._by_service_samplers[self._key(service, env)] = RateSampler(sample_rate)
 
-    def _set_priority(self, span, priority):
-        # type: (Span, int) -> None
-        span.context.sampling_priority = priority
+    def _set_priority(self, span, priority, mechanism, sample_rate):
```

Comment:
I think that would increase the complexity of the implementation? More code to paths to manage, and since we include the samplers as part of the public API it would be hard to remove/merge them in the future?

Reply:
Yeah, it does increase the surface area and allow more codepaths but at the same time it is subtly changing the existing implementation. 

Another benefit to introducing a new sampler instead of mutating is that we can do much easier performance testing.

On the flip side there is so much implementation that is shared so I'm ok with just mutating the existing sampler.